serverlessgo/main.go

```diff
@@ -16,28 +17,76 @@ var (
 )
 
 func main() {
-	fmt.Println("Simulation Started")
 	flag.Parse()
 	
-	arrivalQueue := godes.NewFIFOQueue("arrival")
+	lb := newLoadBalancer()
+	godes.AddRunner(lb)
+
 	poissonDist := &distuv.Poisson{*lambda, rand.NewSource(uint64(time.Now().Nanosecond()))}
 	reqID := int64(0)
 	godes.Run()
 	for godes.GetSystemTime() < duration.Seconds() {
-		arrivalQueue.Place(&request{id: reqID})
+		sendRequestToLoadBalancer(lb, &request{id: reqID, creatingTimestamp: godes.GetSystemTime()})
 		interArrivalTime := poissonDist.Rand()
 		godes.Advance(interArrivalTime)
-		r := arrivalQueue.Get().(*request)
-		r.responseTime = interArrivalTime 
 		reqID++
 	}
 
+	lb.terminate()
 	godes.WaitUntilDone()
-	fmt.Println("Simulation Ended")
+}
+
+type loadBalancer struct {
+	*godes.Runner
+	isTerminated bool
+	arrivalQueue *godes.FIFOQueue
+	arrivalCond *godes.BooleanControl
+}
+
+func newLoadBalancer() *loadBalancer {
+	return &loadBalancer{&godes.Runner{}, false, godes.NewFIFOQueue("arrival"), godes.NewBooleanControl()}
+}
+
+func sendRequestToLoadBalancer(lb *loadBalancer, r *request) {
+	lb.receiveRequest(r)
+}
+
+func (lb *loadBalancer) receiveRequest(r *request) {
+	lb.arrivalQueue.Place(r)
+	lb.arrivalCond.Set(true)
+}
+
+func (lb *loadBalancer) terminate() {
+	lb.arrivalCond.Set(true)
+	lb.isTerminated = true
+}
+
+func (lb *loadBalancer) Run() {
+	fmt.Println("creatingTimestamp,finishingTimestamp,id,status,latency")
+	for {
+		lb.arrivalCond.Wait(true)
+		if lb.arrivalQueue.Len() > 0 {
+			r := lb.arrivalQueue.Get().(*request)
+			r.status = 200
+			r.responseTime = *lambda // temporary value
```

Comment:
Você parece que não precisa do responseTime, certo? Você guarda no arquivo o início e o fim da requisição.

Na real, acho que você nem precisa do finishTime também. Você não usa em lugar nenhum.

Reply:
Resolvi tirar o creatingTimestamp e finishingTimestamp para manter o contrato do documento que criamos.

Reply:
Em relação ao responseTime, não estamos fazendo uso sofisticado ainda. Quando adicionarmos o uso dos arquivos de entrada esse atributo fará mais sentido.